### ISSUE TITLE
Neovide is laggy when certain fonts are used (like Cascadia Code)

### ISSUE THREAD

**Issue by Freemason-12** [neovide/neovide#1687]
**Describe the bug**
When I use Cascadia Code as a default font in my AstroNvim config it starts to lag when scrolling down to the unvisited part of the code or text, if that part was scrolled, going back up is smooth

**To Reproduce**
1. Set Font to Cascadia Code
2. Open some long code with some TreeSitter parser, linter/LSP installed for this language
3. Scroll down
4. See lag

**Expected behavior**
everything smooth

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [KDE Neon 5.26]
 - Neovide Version [e.g. 0.10.3]
 - Neovim Version [e.g. 0.8.1]

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**

**Additional context**

**Comment by shivamchandra75** [comment]
Hey , @Freemason-12 
I am having laggy cursor animations and laggy scroll, with the same environment that you mentioned to Reproduce.
But i found that the jittery animations are noticeable **ONLY WHEN LINTER IS ENABLED**.
I also tried with JetBrainsMono Nerd Font and still laggy animations with linter enabled not so noticeable with linter disabled.
`Note: linter disabled the lag is very little but still there...`

## Video showing lag

https://github.com/neovide/neovide/assets/83809266/8360fa5b-2a65-413d-bbff-925b833df3c6

**Comment by fredizzimo** [comment]
I think that this might have fixed the original issue
* https://github.com/neovide/neovide/pull/2354

The problem @shivamchandra75 has is different, and I think it's caused by some plugins and not by Neovide, so there's nothing we can do about that.

Would it be possible for someone that experienced the lag with Cascadia Code to confirm that it's fixed or not on main (0.12.2 does not include the fix)?

**Comment by fredizzimo** [comment]
@rp1231, are you able to build main from source? There was a bug with loading Cascadia Cove that was fixed here:
* https://github.com/neovide/neovide/pull/2354

If that does not help I guess we need the profiling information according to these instructions:
https://neovide.dev/troubleshooting.html#performance-profiling?

It's ok, if you are not, able to do it, but then we need someone else that can repeat the bug to reproduce it.

**Comment by fredizzimo** [comment]
@rp1231, well, in the profiler I saw one frame that was over 100 ms, because the vsync detection did not trigger, and another frame that took too long to render for no apparent reason. Both of which should be fixed by the d3d frame.

Can you take another profiling log from that branch? 

And to make it easier for me to see what's going on, wait around 10 seconds between before pressing `:`, then around 5 seconds before pressing `esc`, and then 10 seconds again before `:` and so on. Repeat that a few times, so that the lag is visible. Then you can quit, preferably wait 20 seconds before quitting.

If the lag does not happen every time, you can tell how many times you pressed `:` before it happened. 

I can then count the delays to know exactly what's happening.

**Comment by fredizzimo** [comment]
@rp1231, yes, that's also a valid method. I'm not sure why the command did not work though, but maybe there's some difference in the windows version.

**Comment by fredizzimo** [comment]
Thanks. There has to be something going on, on the GPU side, every animation looks roughly like this

![image](https://github.com/neovide/neovide/assets/9946255/21abb2de-607a-43c7-b2ab-96c4ac16a999)

Where the only information we have, is that it's waiting for vblank/vsync for around 150ms. The rest of the frames are a stable 60 FPS. 

Which graphics card do you have, I can only see the CPU information in the trace?

The bad thing is that I'm not sure how to track this down further without using complicated tools, which furthermore takes big captures.

There's one thing you can try, but I'm not sure if it works properly, add `--features gpu_profiling` to the command. This will slow everything down, but the trace could still give useful information. At least if the GPU is busy or not.

**Comment by fredizzimo** [comment]
The trace looks very strange. I guess I have to compare how it looks on my system to know what's going on. 

But it might just be that the gpu profiling slows it down so much that it becomes useless. Maybe it would be good to have one with the same parameters and no `gpu_trace` as well, so I have all combinations when I start to compare.

**Comment by fredizzimo** [comment]
Thanks. I still have no idea what's going on, but there's clearly something strange, you should be getting around 1000 FPS, I think, but it's not even hitting a stable 60.

But that might be a problem with my d3d branch, I'm not sure it can properly run without vsync. The standard master version with opengl should do that though.

**Comment by fredizzimo** [comment]
This is very weird, first it runs at around 1000FPS, with a few frames taking long around 3 ms,  then it throttles to 60 FPS, and then to around 6FPS (150 ms) for a frame, and then back to 60 FPS.  And it's blocking in `wait_for_vsync` which is just a sleep. I think this might have something to do with the timer resolution throttling https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-setprocessinformation. But I need to read up more on that.

The `--no-idle` behaviour also appears to be slightly broken and I don't understand how it runs at 30 FPS, if you set `:let g:neovide_refresh_rate=9999`. But that might be another bug, and not related to the problem
